packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2269,14 +2273,16 @@ class RenderFittedBox extends RenderProxyBox {
       _resolve();
       final Size childSize = child.size;
       final FittedSizes sizes = applyBoxFit(_fit, childSize, size);
-      final double scaleX = sizes.destination.width / sizes.source.width;
-      final double scaleY = sizes.destination.height / sizes.source.height;
+      final double scaleX = _div_handle_zero(sizes.destination.width, sizes.source.width);
```

Comment:
This seems like it should be `assert(sizes.source.width != 0)`, and maybe `assert(sizes.source.width.isFinite)`, no?

Reply:
That's a possible solution too. We'd have to modify `test/rendering/proxy_box_test.dart: RenderFittedBox paint` to not send a 0 source width though. Let me see how that test can be modified.

Reply:
Is it because it's sending in `Size.zero` for constraints?

Why did this test pass before but fail now? Your change doesn't seem to have anything to do with it.

Reply:
(It does look like the intention here is to have a size.zero be a no-op though)